streams/src/test/java/org/apache/kafka/streams/state/internals/TimestampedWindowStoreBuilderTest.java

```diff
@@ -179,4 +179,9 @@ public void shouldThrowNullPointerIfTimeIsNull() {
         new TimestampedWindowStoreBuilder<>(supplier, Serdes.String(), Serdes.String(), null);
     }
 
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerIfMetricsScopeIsNull() {
```

Comment:
Where do we pass a `null` value? Also, where is the `MetricScope` in this test?

Reply:
`supplier#metricsScope()` returns `null`, for its `setUp()` configuration. It is same to `[KeyValueStoreBuilderTest, TimestampedKeyValueStoreBuilderTest]#shouldThrowNullPointerIfMetricsScopeIsNull`. `SessionStoreBuilderTest#shouldThrowNullPointerIfMetricsScopeIsNull` does same, but it also validates Exception message. (Wait, is this way more recommended?)

Reply:
Thanks for the clarification. However, I am still not sure if I understand this test. The constructor `TimestampedWindowStoreBuilder` does not call `supplier.metricsScope()` -- hence, it am still puzzled why this test would throw a NPE? To test `metricScope()` we would need to call `build()` method. In fact, following the call trace, we would never check if `metricScope()` returns `null` or not. Seems worth to fix... (maybe add a check in `MeteredWindowStore` constructor? -- not sure if the other `MeteredXxxStore` have a similar check. If not, would be good to add one.)

Seems it comes from `supplier.name()` -- for this case, the test setup is incorrect -- also in the test above actually, that would also fail with `supplier.name()` NPE, but not with the NPE they want to test.

> does same, but it also validates Exception message. (Wait, is this way more recommended?)

Yes, it's always recommended to verify the exception message to make sure we test the right think. Using `(expected = NullPointerException.class)` annotation is not ideal.

Reply:
@mjsax I reviewed the problem more carefully and concluded that you are right - the `NullPointerException` here is not related to `MetricScope`. We need to rewrite the series of tests.

Let's review `TimestampedWindowStoreBuilderTest#shouldThrowNullPointerIfMetricsScopeIsNull` as an example. Here, `supplier#name` is `null` so when instantiating `TimestampedWindowStoreBuilder`, it fails in its parent class's constructor, `AbstractStoreBuilder` line 41 - `Objects.requireNonNull(name, "name cannot be null");`. It is also why `SessionStoreBuilderTest#shouldThrowNullPointerIfMetricsScopeIsNull` ends with a error message "name cannot be null" - obviously, it also has nothing to do with `MetricScope`.

Here is the debugger status; As you can see, `supplier#name` is `null` here.

![name-null](https://user-images.githubusercontent.com/2375128/79251817-a75b8b00-7ebb-11ea-9b1e-550d203c82df.png)

I am now rewriting the tests. Stay tuned! :smile:

Reply:
Thanks! Highly appreciated!